### PR TITLE
Added Invariant Culture Warning option

### LINKED ISSUE #899: Add option to disable invariant culture warning
Currently a warning message is reported if a pipeline is executed in a invariant culture environment. Sometimes an invariant culture environment is required and this message is not required.

For these cases we should provide an option to disable the warning.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #899.

Added `Execution.InvariantCultureWarning` option.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Code changes**
  - [x] Have unit tests created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Microsoft/PSRule/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by ArmaanMcleod** [comment]
@BernieWhite This should be ready for review. 

I tested this option across some existing tests to suppress the warning, seems to be working fine.

Let me know what you think. I temporarily hadto enable `Output.Verbosity = 'Detailed'` to figure out which pester tests were showing the warning. Can reset this if you want if its too verbose for general use.

File: docs/concepts/PSRule/en-US/about_PSRule_Options.md [hunk 1]
```diff
@@ -916,6 +917,52 @@ variables:
   value: false
 ```
 
+### Execution.InvariantCultureWarning
+
+When evaluating rules inside a Linux CI host, if invariant culture is used, a warning is shown by default.
```
Comment (ArmaanMcleod) [inline docs/concepts/PSRule/en-US/about_PSRule_Options.md:922]: I did notice that this warning can also show up in Windows CI, so maybe I should omit `Linux` here?

File: src/PSRule/Configuration/ExecutionOption.cs [hunk 2]
```diff
@@ -120,6 +127,12 @@ internal static ExecutionOption Combine(ExecutionOption o1, ExecutionOption o2)
         [DefaultValue(null)]
         public bool? SuppressedRuleWarning { get; set; }
 
+        /// <summary>
+        /// Determines if warning is raised when invariant culture us used.
```
Comment (BernieWhite) [inline src/PSRule/Configuration/ExecutionOption.cs:131]:
```suggestion
        /// Determines if warning is raised when invariant culture is used.
```

**Comment by BernieWhite** [comment]
> @BernieWhite This should be ready for review.
> 
> I tested this option across some existing tests to suppress the warning, seems to be working fine.
> 
> Let me know what you think. I temporarily had to enable `Output.Verbosity = 'Detailed'` to figure out which pester tests were showing the warning. Can reset this to `Normal` if you want if its too verbose for general use.

This should be fine in CI. Might be good to limit it in VSCode though.

File: docs/concepts/PSRule/en-US/about_PSRule_Options.md (hunk 1 above)
Comment (BernieWhite) [reply docs/concepts/PSRule/en-US/about_PSRule_Options.md:922]: Good idea, let's not specifically mention Linux.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice. A few comments but otherwise good to merge.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Nice. A few comments but otherwise good to merge.

Thanks @BernieWhite. 

I made a small change to add `Detailed` output for CI(AzDO and Github actions). Let me known if your ok with this 🙂.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod All good.

**Pull request merged**